kore/src/Kore/Exec.hs

```diff
@@ -274,15 +274,23 @@ prove
     -- ^ The spec module
     -> smt (Either (TermLike Variable) ())
 prove limit definitionModule specModule =
-    evalSimplifier definitionModule $ initialize definitionModule $ \initialized -> do
+    evalSimplifier definitionModule
+    $ initialize definitionModule
+    $ \initialized -> do
         tools <- Simplifier.askMetadataTools
         let Initialized { rewriteRules } = initialized
             specClaims =
                 map
                     (Bifunctor.second $ expandOnePathSingleConstructors tools)
                     (extractOnePathClaims specModule)
+            mapMSecond :: Monad m => (b -> m [c]) -> (a, b) -> m [(a, c)]
```

Comment:
Do I understand correctly: This takes a function, which simplifies the left-hand side of a rule into a disjunction, and returns all of the rules that produces, given a single left- and right-hand side?